### PR TITLE
app: sign with Capella domain

### PR DESCRIPTION
**Pull request by gsora**
Following EIP-7044[1], sign exits with Capella domain for perpetual validity.

[1]: https://eips.ethereum.org/EIPS/eip-7044

category: bug
ticket: none

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -400,16 +452,31 @@ func signExit(ctx context.Context, eth2Cl eth2wrap.Client, valIdx eth2p0.Validat
...
+	forkHash string,
+	spec map[string]any,
+) ([32]byte, error) {
 	sigRoot, err := exit.HashTreeRoot()
 	if err != nil {
 		return [32]byte{}, errors.Wrap(err, "exit hash tree root")
 	}
 
-	domain, err := signing.GetDomain(ctx, eth2Cl, signing.DomainExit, exitEpoch)
+	domainType, ok := spec["DOMAIN_VOLUNTARY_EXIT"]
```
Comment (LukeHackett12) [inline app/app.go:467]: As this is the only field used will we pass the domain type at the top and error earlier?

File: app/util/util.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
```
Comment (LukeHackett12) [inline app/util/util.go:2]:
```suggestion
```

File: app/util/util_test.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
```
Comment (LukeHackett12) [inline app/util/util_test.go:2]:
```suggestion
```

**Review by LukeHackett12 — APPROVED** [review]
formatting and style comments, otherwise I like it 💯

File: app/app.go [hunk 4]
```diff
@@ -144,14 +155,16 @@ func Run(ctx context.Context, config Config) error {
...
 					oAPI,
 					cl,
 					signedExits,
 					fetchedSignedExits,
 					config.EjectorExitPath,
 					shareIdx,
 					identityKey,
+					genesis.Data.GenesisValidatorsRoot,
+					capellaForkHash,
+					specResp.Data,
```
[the remark this answers is not in the document]
Comment (gsora) [reply app/app.go:167]: Yeah that's what happens with Go dependency injection (passing arguments).

It's okay if we do this later down the line?

File: app/app.go (hunk 1 above)
Comment (gsora) [reply app/app.go:467]: we might, but from a Go standpoint it's really the same thing, `map`s are references anyway

**Review by LukeHackett12 — APPROVED** [review]

**Pull request merged**